tiledb/sm/serialization/array.cc

```diff
@@ -147,13 +147,28 @@ Status array_to_capnp(
         *(schema.second.get()), &schema_builder, client_side));
   }
 
-  if (array->serialize_non_empty_domain()) {
+  if (array->use_refactored_array_open()) {
+    if (array->serialize_non_empty_domain()) {
+      auto nonempty_domain_builder = array_builder->initNonEmptyDomain();
+      RETURN_NOT_OK(
+          utils::serialize_non_empty_domain(nonempty_domain_builder, array));
+    }
+
+    if (array->serialize_metadata()) {
+      auto array_metadata_builder = array_builder->initArrayMetadata();
+      // If this is the Cloud server, it should load and serialize metadata
+      // If this is the client, it should have previously received the array
+      // metadata from the Cloud server, so it should just serialize it
+      Metadata* metadata = nullptr;
+      // Get metadata. If not loaded, load it first.
+      RETURN_NOT_OK(array->metadata(&metadata));
+      RETURN_NOT_OK(metadata_to_capnp(metadata, &array_metadata_builder));
+    }
+  } else {
```

Comment:
For the else clause below, do we need a check for if the nonempty domain and array metadata have been loaded? If this is the client submitting a query in the non-refactor (v1) open. Then won't the below serialize the metadata that is empty?

Reply:
Agreed, I also noticed that but that was the existing behavior for array open v1 before v2 was introduced so I didn't dare to change it. For metadata, it would currently serialize empty metadata if not loaded, and for non_empty_domain it would force loading it - even more inconsistent.
I changed it to check if the data is loaded for both of them, plese let me know if that's what you'd expect.

Reply:
I think this logic is correct, this looks good to me! Thank you for adjusting it.